src/pandroid/app/src/main/java/com/panda3ds/pandroid/data/SMDH.java

```diff
@@ -0,0 +1,167 @@
+package com.panda3ds.pandroid.data;
+
+import android.graphics.Bitmap;
+
+import com.panda3ds.pandroid.data.game.GameRegion;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+public class SMDH {
+    private static final int ICON_SIZE = 48;
+    private static final int META_OFFSET = 0x8 ;
+    private static final int META_REGION_OFFSET = 0x2018;
+    private static final int IMAGE_OFFSET = 0x24C0;
+
+    private int metaIndex = 1;
+    private final ByteBuffer smdh;
+    private final String[] title = new String[12];
+    private final String[] publisher = new String[12];
+    private final int[] icon;
+
+    private final GameRegion region;
+
+    public SMDH(byte[] source){
+        smdh = ByteBuffer.allocate(source.length);
+        smdh.position(0);
+        smdh.put(source);
+        smdh.position(0);
+
+        region = parseRegion();
+        icon = parseIcon();
+        parseMeta();
+    }
+
+    private GameRegion parseRegion(){
+        GameRegion region;
+        smdh.position(META_REGION_OFFSET);
+
+        int regionMasks = smdh.get() & 0xFF;
+
+        final boolean japan = (regionMasks & 0x1) != 0;
+        final boolean northAmerica = (regionMasks & 0x2) != 0;
+        final boolean europe = (regionMasks & 0x4) != 0;
+        final boolean australia = (regionMasks & 0x8) != 0;
+        final boolean china = (regionMasks & 0x10) != 0;
+        final boolean korea = (regionMasks & 0x20) != 0;
+
+        final boolean taiwan = (regionMasks & 0x40) != 0;
```

Comment:
metaIndex should be initialized here instead of in the class to avoid bugs later down the line
Also the region booleans don't need to be finals since they're local variables instead of class variables.
Taiwan should also be with the other regions

Reply:
metaIndex should probably also use proper named constants instead of arbitrary numbers